qcodes/dataset/descriptions.py

```diff
@@ -0,0 +1,107 @@
+import io
+from typing import Dict, Any
+import json
+
+from qcodes.dataset.dependencies import InterDependencies
+
+
+class RunDescriber:
+    """
+    The object that holds the description of each run in the database. This
+    object serialises itself to a string and is found under the run_description
+    column in the runs table
+
+    Extension of this object is planned for the future, for now it holds the
+    parameter interdependencies. Extensions should be objects that can
+    serialize themselves added as attributes to the RunDescriber , such that
+    the RunDescriber can iteratively serialize its attributes when serializing
+    itself.
+    """
+
+    def __init__(self, interdeps: InterDependencies) -> None:
+
+        if not isinstance(interdeps, InterDependencies):
+            raise ValueError('The interdeps arg must be of type: '
+                             f'InterDependencies. Got {type(interdeps)}.')
+
+        self.interdeps = interdeps
+
+    def serialize(self) -> Dict[str, Any]:
+        """
+        Serialize this object into a dictionary
+        """
+        ser = {}
+        ser['Parameters'] = self.interdeps.serialize()
+        return ser
+
+    @classmethod
+    def deserialize(cls, ser: Dict[str, Any]) -> 'RunDescriber':
+        """
+        Make a RunDescriber object based on a serialized version of it
+        """
+        idp = InterDependencies.deserialize(ser['Parameters'])
+        rundesc = cls(interdeps=idp)
+
+        return rundesc
+
+    @staticmethod
+    def _ruamel_importer():
+        try:
+            from ruamel_yaml import YAML
```

Comment:
the module is loaded for every time `to_yaml`/`from_yaml` is called. does it take a considerable amount of time to load the yaml module? if so, this could be contributing significantly to the overall speed of `to_yaml`/`from_yaml` methods. does it make sense to put the imports out of the class, and define a flag `yaml_exists` or smth (or set YAML to None if it was not possible to load it) and use it in this method in order to improve performance? I also thought that it is generally a good practice to do all the imports on top (unless really necessary).